src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +279,215 @@ def new_component(
     except OSError as ose:
         print("[ERROR] {}".format(ose))
     return 1
+
+
+def is_valid_name(word):
+    invalid_characters = [
+        "#",
+        "%",
+        "&",
+        "{",
+        "}",
+        "/",
+        "\\",
+        "<",
+        ">",
+        "*",
+        "?",
+        " ",
+        "$",
+        "!",
+        "'",
+        '"',
+        ":",
+        "@",
+        "+",
+        "`",
+        "|",
+        "=",
+    ]
+    for char in invalid_characters:
+        if char in word:
+            return char
+    return "valid"
+
+
+def get_port_input(namespace):
+    # Gather inputs to use as context for the port template
+    defaults = {
+        "port_name": "ExamplePort",
+        "short_description": "Example usage of port",
+        "dir_name": ".",
+        "namespace": namespace,
+        "arg_list": [],
+    }
+    valid_name = False
+    valid_dir = False
+    valid_namespace = False
+    args_done = False
+    arg_list = []
+    while not valid_name:
```

Comment:
optional recommendation: You might be able to simplify the code and remove some duplication by refactoring the user input prompt into a function.

Reply:
I agree, this could be much simpler